kpops/component_handlers/kafka_connect/model.py

```diff
@@ -123,3 +125,10 @@ class KafkaConnectorResetterConfig(CamelCaseConfigModel):
 class KafkaConnectorResetterValues(HelmAppValues):
     connector_type: Literal["source", "sink"]
     config: KafkaConnectorResetterConfig
+    image_tag: str = Field(default="latest")
+
+    @pydantic.field_validator("image_tag", mode="before")
```

Comment:
wouldn't after make more sense? at least then we already know it's a str after Pydantic is done with its checks. Alternatively, I believe we could even set the regex pattern directly on the Pydantic field.

Reply:
e.g.

```python
ImageTag = Annotated[str, StringConstraints(pattern=IMAGE_TAG_PATTERN)]
```

Reply:
Wouldn't that be enough?
```
    image_tag: str = Field(
        default="latest",
        pattern=IMAGE_TAG_PATTERN,
        description=describe_attr("image_tag", __doc__),
    )
```

Reply:
yes, that also works, but I believe `Annotated` is the newer/recommended way. I don't mind either way

Reply:
perhaps Annotated is transported to the generated schema, it might be worth to check

Reply:
Changing it to Annotated, I get an error on `gen-schema` pre-commit hook 🤔 
```
pydantic.errors.PydanticUserError: `PipelineSchema` is not fully defined; you should define `pydantic`, then call `PipelineSchema.model_rebuild()`.

For further information visit https://errors.pydantic.dev/2.6/u/class-not-fully-defined

```

Reply:
idk, for me both work and produce the exact same schema

Reply:
After struggling with the Annotated way I went for the classical `pattern=` input in the `Field` method. Updated the tests accordingly